src/components/Auth/SignUp.jsx

```diff
@@ -0,0 +1,197 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { NavLink, Link } from 'react-router-dom';
+import { isEmail } from 'validator';
+import { handleSignUpForm, submitSignUp } from '../../redux/actions';
+import Button from '../common/Button/Button';
+import Input from '../common/Input/Input';
+import fb from '../../assets/images/facebook.png';
+import twitter from '../../assets/images/twitter.png';
+import google from '../../assets/images/google.png';
+
+export class SignUp extends Component {
+  state = {
+    validEmail: true,
+    emailError: 'Email is not valid',
+    validPassword: true,
+    passwordError: 'Password must have at least 6 characters',
+    validUsername: true,
+    usernameError: 'username is not allowed to be empty',
+  };
+
+  onSubmitButton = () => {
+    const {
+      onSubmit, email, username, password,
+    } = this.props;
+    this.setState({ validEmail: true, validPassword: true, validUsername: true });
+
+    if (username.length < 1) {
+      this.setState({
+        validUsername: false,
+        usernameError: 'username is not allowed to be empty',
+      });
+      return;
+    }
+
+    if (!isEmail(email)) {
+      this.setState({ validEmail: false });
+      return;
+    }
+    if (password.length < 5) {
```

Comment:
empty line on line 41

Reply:
I am going to remove it in the next commit!